### ISSUE TITLE
Passing "profile" or "profileName" to workbench.action.terminal.newWithProfile doesn't work

### ISSUE THREAD

**Issue by AdamPacanowski** [microsoft/vscode#126738]
Issue Type: <b>Bug</b>

TL;DR Passing information any information profile to workbench.action.terminal.newWithProfile doesn't work.

### Steps to reproduce
1. Prepare some integrated terminal profiles in `settings.json` file e.g.
  ```{
      "terminal.integrated.profiles.windows": {
          "PowerShell": {
            "source": "PowerShell",
            "icon": "terminal-powershell"
          },
          "CommandPrompt": {
            "path": [
              "${env:windir}\\Sysnative\\cmd.exe",
              "${env:windir}\\System32\\cmd.exe"
            ],
            "args": [],
            "icon": "terminal-cmd"
          },
          "Git Bash": {
            "source": "Git Bash"
          }
        },
  }
  ```
2. Set custom keybindings e.g.
```
[
    {
        "key": "alt+x",
        "command": "workbench.action.terminal.newWithProfile",
        "args": {
            "profileName": "Git Bash" // it can be "CommandPrompt" or "PowerShell" it doesn't matter
        }
      }
]
```
3. Press "alt+x". You will get default profile.

### Other remarks
When you see there:
https://github.com/microsoft/vscode/blob/b2614ce5042b8711d146ea83d67493482ccf0a49/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L136 you can think that if you define "full" profile (not only reference to some profile in settings.json) it could work. - No, it doesn't work (You get standard prompt.)
Accordingly to that fragment: https://github.com/microsoft/vscode/blob/b2614ce5042b8711d146ea83d67493482ccf0a49/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L146 if vscode doesn't find `profileName` property, it will display prompt which is OK, but it looks like vscode doesn't searching through define profiles.

If I am wrong, please correct me, because not only me have problem with that command (https://stackoverflow.com/questions/67734533/how-to-add-keybinding-to-open-specific-terminal-profile-in-vscode).

VS Code version: Code - Insiders 1.58.0-insider (7dea6d7402b55671410a129661cad24b55aed9f4, 2021-06-18T13:58:43.903Z)
OS version: Windows_NT x64 10.0.19041
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7440HQ CPU @ 2.80GHz (4 x 2808)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.88GB (0.95GB free)|
|Process Argv|--crash-reporter-id ad0143d5-41e2-4747-92c3-402d90af2415|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
pythonvspyt602:30291494
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30291487
pythontb:30258533
pythonvspyt551:30291412
vspre833:30321513
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscus158:30321503
pythonvsuse255:30319630
vscorehov:30301224
vscod805cf:30301675
pythonvspyt200:30323110
vscextlang:30310088
vsccppwt:30312692

```

</details>

**Comment by Tyriar** [comment]
Thanks for the investigation. This indeed does appear to need a full profile so we should:

- Correct the type received in the action to indicate it may only be the profile name https://github.com/microsoft/vscode/blob/cae22b8826afa58be54e99bd5c49e6340441616c/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L147-L147
- Add a proper schema for newWithProfile https://github.com/microsoft/vscode/blob/cae22b8826afa58be54e99bd5c49e6340441616c/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L137-L143
- Resolve the profile before calling `createTerminal` if it's only a name as it requires a full profile, not just the name
- Consider changing the `createTerminal` interface such that this type of accident can be avoided in the future. Right now both of these are valid so essentially anything can be passed in `createTerminal()`, `createTerminal({})`
